src/meta/app/src/principal/user_auth.rs

```diff
@@ -158,6 +158,21 @@ impl AuthInfo {
         AuthInfo::new(auth_type, auth_string, need_change)
     }
 
+    pub fn create_set_need_change(&self, need_change: bool) -> AuthInfo {
```

Comment:
why not use create2?

Reply:
It like a alter impl not create.

Reply:
we don't change `auth_type` and `auth_string`, only set `need_change`, the logic is different with `create2`.